samples/drivers/crypto/sample.yaml

```diff
@@ -27,3 +27,9 @@ tests:
         - ".*: CBC Mode"
         - ".*: CTR Mode"
         - ".*: CCM Mode"
+  sample.drivers.crypto.stm32:
+    tags: crypto
+    platform_whitelist: mikroe_mini_m4_for_stm32
+    filter: dt_compat_enabled("st,stm32-cryp")
```

Comment:
Why do we need both `filter` and `platform_whitelist`?

Reply:
Removed filter.

Reply:
@galak, @ulfalizer genuine question: how is `dt_compat_enabled` supposed to work as sanitycheck filter?
When running sanitycheck, do we build all boards.dts to see if "compat,foo' is actually enabled ? (This is not what I'm seeing at least)
Or is the filter applied after platforms builds and then if filter does not match, the configuration is skipped? (which would explain the mysterious "skipped").

In any case, it seems that this filter is missing something as `sample.drivers.crypto.stm32` is applied on bunch of other platforms ....